src/app/frontend/common/pipes/safehtml.ts

```diff
@@ -15,17 +15,41 @@
 import {Pipe, SecurityContext} from '@angular/core';
 import {DomSanitizer, SafeHtml} from '@angular/platform-browser';
 
+const ansiColorClass = require('ansi-to-html');
+const ansiColor = new ansiColorClass();
+
+enum TextMode {
+  Default = 'Default',
+  Colored = 'Colored',
+}
+
 /**
  * Formats the given value as raw HTML to display to the user.
  */
 @Pipe({name: 'kdSafeHtml'})
 export class SafeHtmlFormatter {
   constructor(private readonly sanitizer: DomSanitizer) {}
 
-  transform(value: string): SafeHtml {
-    return this.sanitizer.sanitize(
+  transform(value: string, mode: TextMode = TextMode.Default): SafeHtml {
+    let result: SafeHtml = null;
+    let content = this.sanitizer.sanitize(
       SecurityContext.HTML,
       value.replace('<', '&lt;').replace('>', '&gt;'),
     );
+
+    // Handle conversion of ANSI color codes.
+    switch (mode) {
+      case TextMode.Colored:
+        content = ansiColor.toHtml(content.replace(/&#27;/g, '\u001b'));
+        result = this.sanitizer.bypassSecurityTrustHtml(content);
```

Comment:
I am worried about using `bypassSecurityTrustHtml`. This might lead to security issues and allow execution of 3rd party code through logs.

Reply:
All of the HTML was sanitized in the prior step, the only new HTML would be that added by the ANSI color.  I believe this to be fully safe at and that point.